src/data/nat/basic.lean

```diff
@@ -977,19 +977,27 @@ by unfold bodd div2; cases bodd_div2 n; refl
 section
 variables {α : Sort*} (op : α → α)
 
-@[simp] theorem iterate_zero (a : α) : op^[0] a = a := rfl
+@[simp] theorem iterate_zero : op^[0] = id := rfl
```

Comment:
Maybe state `iterate_zero_apply`, to save one rewrite sometimes.

Reply:
Done.